cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -738,7 +738,13 @@ Status TransferInt96(RecordReader* reader, MemoryPool* pool,
   RETURN_NOT_OK(::arrow::AllocateBuffer(pool, length * sizeof(int64_t), &data));
   auto data_ptr = reinterpret_cast<int64_t*>(data->mutable_data());
   for (int64_t i = 0; i < length; i++) {
-    *data_ptr++ = Int96GetNanoSeconds(values[i]);
+    if (values[i].value[2] == 0) {
```

Comment:
I'm not sure I understand why 0 is an invalid value (I guess this is specific semantics of int96  type)?

Reply:
It's not invalid per se, it just doesn't convert to a valid `int64_t` timestamp.
A better fix might to be to detect all non-convertible values, and raise an error or use a placeholder, but that's overkill for a deprecated feature IMHO.

Reply:
ah, OK.